### PR TITLE
bumps mysql connector version and extracts var in Dockerfile

### PR DESCRIPTION
**Pull request by josemigallas**

### CODE REVIEW

File: wildfly/Dockerfile [hunk 1]
```diff
@@ -10,9 +10,10 @@ RUN yum install -y unzip wget && yum -q clean all
 
 ## MYSQL
 ENV mysql_module_dir=$JBOSS_HOME/modules/com/mysql/jdbc/main/
+ENV mysql_connector_jar=mysql-connector-java.jar
 RUN mkdir -p ${mysql_module_dir}
-RUN wget -O mysql-connector-java-5.1.18.jar http://search.maven.org/remotecontent\?filepath\=mysql/mysql-connector-java/5.1.18/mysql-connector-java-5.1.18.jar
-RUN mv mysql-connector-java-5.1.18.jar ${mysql_module_dir}
+RUN wget -O ${mysql_connector_jar} http://search.maven.org/remotecontent\?filepath\=mysql/mysql-connector-java/5.1.44/mysql-connector-java-5.1.44.jar
```
Comment (matzew) [inline wildfly/Dockerfile]: let's keep the version number, on the filename, for explicity

File: wildfly/configuration/xml/mysql-module.xml [hunk 2]
```diff
@@ -14,7 +14,7 @@
 -->
 <module xmlns="urn:jboss:module:1.0" name="com.mysql.jdbc">
   <resources>
-    <resource-root path="mysql-connector-java-5.1.18.jar"/>
+    <resource-root path="mysql-connector-java-5.1.44.jar"/>
```
Comment (matzew) [inline wildfly/configuration/xml/mysql-module.xml]: I've not actually run this, but I think you are not downloading it as `mysql-connector-java-5.1.44.jar`, but as `mysql-connector-java.jar` instead, hence this reference would be invalid - no?

Comment (josemigallas) [reply wildfly/configuration/xml/mysql-module.xml]: I see, that's true. This is related with your other comment above, for more explicitness it's good to have the version hardcoded, however we would have the version in 3 different places, making updates a bit "harder". This very PR, with 3 places to change, I made already a couple of mistakes. Do you then still agree with having the version hardcoded?

Comment (matzew) [reply wildfly/configuration/xml/mysql-module.xml]: yes, version numbers are extremely important metadata - at least to me

Comment (josemigallas) [reply wildfly/configuration/xml/mysql-module.xml]: done

**Comment by matzew** [comment]
:+1:

**Pull request merged**